src/clib/pio_darray_int.c

```diff
@@ -394,10 +417,16 @@ int write_darray_multi_par(file_desc_t *file, int nvars, int fndims, const int *
                 break;
 #endif
             default:
-                return pio_err(ios, file, PIO_EBADIOTYPE, __FILE__, __LINE__,
+                ierr = pio_err(ios, file, PIO_EBADIOTYPE, __FILE__, __LINE__,
```

Comment:
It would be better to put a break for the default case.

Reply:
This should be fixed now